cstar/roms/input_dataset.py

```
@@ -53,7 +121,9 @@ def __repr__(self) -> str:
 
         return repr_str
 
-    def partition(self, np_xi: int, np_eta: int):
+    def partition(
+        self, np_xi: int, np_eta: int, overwrite_existing_files: bool = False
```

I was going to add a comment that, if `overwrite_existing_files is True`, we should maybe clear out the old files just to make sure we don't end up with a confusing state where some are overwritten but some others may be left over from the previous partitioning. But, now I'm wondering if we should just not support overwriting here at all, and make people manually clean up the old files if they're going to do so? Thoughts?

---

I considered this, though it is just a tidying thing: if, e.g. we first partition into 25 tiles and then 15, only the first 15 will be used, so 16-25 will just be hanging around doing nothing, in an ugly but ultimately harmless way. I also considered moving the existing files to a temporary location and having a `finally` to return them in case of failure, which covers both cases. WDYT?

---

Yeah, I just imagine it will be easily forgotten and someone will look at the directly later and see 25 files, and not know that 16-25 are now totally incompatible with 1-15.

I guess the question to me is: how often is someone going to hit this "I want to intentionally change partitioning" condition? If it's often enough that we should handle it for them, we should handle it well. My assumption (which could be wrong) is that it will be rare enough that we can just throw an error for now and tell them to clean out the old files themselves.

---

Agree, I just took the "well they shouldn't be peeking at the files, C-Star is taking care of it for them" line. But either way it's a bit gross and not completely insane to do the temporary move thing, so I'll just add that.

---

[8ed114f](https://github.com/CWorthy-ocean/C-Star/pull/252/commits/8ed114f075c136225a9ff46911fe466f7dccad4f)